design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,471 @@
+| [Deployment](*https://kubernetes.io/docs/concepts/workloads/controllers/deployment/*) | K8s entity that ensures multiple replications of a pod are running all the time. If a pod terminates, another one will be created. |
+| [Kubernetes Helm](https://helm.sh/)                          | A tool that streamlines and organizes the management of Kubernetes installation and deployment processes. |
+| [Helm Chart](https://helm.sh/docs/topics/charts/)            | A collection of files that describe a related set of Kubernetes resources. A chart can be used to describe a simple pod of a complex application. |
+
+## Useful links
+
+| Name                | Link                                                         |
+| ------------------- | ------------------------------------------------------------ |
+| Phase 2 Feature Doc | https://github.com/cyberark/secrets-provider-for-k8s/issues/102 |
+
+## Background
+
+The Cyberark Secrets Provider for K8s enables customers to use secrets stored and managed in Conjur/DAP vault and consume them as K8s Secrets for their application containers.
+
+During Phase 1, an *init container* was deployed in the same Pod as the customer's applications to populate the K8s secrets defined in the pod application's manifest.
+
+### Motivation
+
+The current implementation of the Secrets Provider only supports one application and makes the deployment of the Secrets Provider deeply coupled to the customer's applications. 
+
+## Requirements
+
+The solution should support the following capabilities:
+
+###  Milestone 1 *(current)*
+
+- Secrets Provider runs as a separate entity, serving multiple application containers that run on multiple pods
+- Solution needs to be native to Kubernetes
+- Lifecycle should support update/removal of deployment
+- Provide a way for customer to understand the state of the Secret Provider - when it finished initializing
+
+###  Milestone 2
+
+- Secrets Provider support secret rotation
+
+For seamless shift between milestones, we will deploy using Helm.
+
+## Solution
+
+The solution is to allow different deployments and behave differently based on the use-case chosen. The following decision tree depicts customer uses-cases for Secrets Provider deployments:
+
+ ![Secrets Provider flavors decision flow chart](https://user-images.githubusercontent.com/31179631/85747023-975bf500-b70f-11ea-8e26-1134068fe655.png) 
+
+These variations will be supported using the same Secrets Provider image, but the behavior will vary dynamically depending on the chosen deployment method that is configured via Helm Chart.
+
+*In this document we will focus on **Milestone 1***, deploying the Secrets Provider as a Job
+
+### Milestone 1: Serve K8s secrets to multiple applications once (no rotation)
+
+### Design
+
+Customer will install our **Helm Chart** to deploy the Secrets Provider as a **Job**. Once installed, the Job spins up and authenticates to Conjur/DAP via authn-k8s.
```

Comment:
What about the init container? as we add HELM we should enable deploying as init container as well to better support our current users

Reply:
deploying an init container with Helm?

Reply:
not saying it's not possible but how can it be done when we don't know how the application container looks like?

Reply:
deploying init container in helm is not a common thing to do and also problematic technically.
I dont think we should do it